tests/token/erc20/test_ERC20_Upgradeable.py

```diff
@@ -61,134 +59,109 @@ async def token_init(contract_defs):
-    admin, _, _, _, proxy = token_factory
+    admin, _, proxy, *_ = token_factory
 
     await signer.send_transaction(
         admin, proxy.contract_address, 'initializer', [
-            NAME,
-            SYMBOL,
-            DECIMALS,
-            *INIT_SUPPLY,
-            admin.contract_address,
-            admin.contract_address
+            NAME,                       # name
+            SYMBOL,                     # symbol
+            DECIMALS,                   # decimals
+            *INIT_SUPPLY,               # initial supplt
+            admin.contract_address,     # recipient
+            admin.contract_address      # admin
         ])
 
-    # check name
-    execution_info = await signer.send_transaction(
-        admin, proxy.contract_address, 'name', [])
-    assert execution_info.result.response == [NAME]
-
-    # check symbol
-    execution_info = await signer.send_transaction(
-        admin, proxy.contract_address, 'symbol', []
-    )
-    assert execution_info.result.response == [SYMBOL]
-
-    # check decimals
-    execution_info = await signer.send_transaction(
-        admin, proxy.contract_address, 'decimals', []
+    execution_info = await signer.send_transactions(
+        admin,
+        [
+            (proxy.contract_address, 'name', []),
+            (proxy.contract_address, 'symbol', []),
+            (proxy.contract_address, 'decimals', []),
+            (proxy.contract_address, 'totalSupply', [])
+        ]
     )
-    assert execution_info.result.response == [DECIMALS]
 
-    # check total supply
-    execution_info = await signer.send_transaction(
-        admin, proxy.contract_address, 'totalSupply', []
-    )
-    assert execution_info.result.response == [*INIT_SUPPLY]
+    # check values
+    expected = [NAME, SYMBOL, DECIMALS, *INIT_SUPPLY]
+    assert execution_info.result.response == expected
 
 
 @pytest.mark.asyncio
 async def test_upgrade(after_initializer):
-    admin, _, _, token_v2, proxy = after_initializer
+    admin, _, proxy, _, token_v2 = after_initializer
 
     # transfer
     await signer.send_transaction(
-        admin, proxy.contract_address, 'transfer', [
-            USER,
-            *AMOUNT
-        ]
+        admin, proxy.contract_address, 'transfer', [USER, *AMOUNT]
     )
 
     # upgrade
     await signer.send_transaction(
-        admin, proxy.contract_address, 'upgrade', [
-            token_v2.contract_address
-        ]
+        admin, proxy.contract_address, 'upgrade', [token_v2.class_hash]
     )
 
-    # check admin balance
-    execution_info = await signer.send_transaction(
-        admin, proxy.contract_address, 'balanceOf', [
-            admin.contract_address
+    # fetch values
+    execution_info = await signer.send_transactions(
+        admin,
+        [
+            (proxy.contract_address, 'balanceOf', [admin.contract_address]),
+            (proxy.contract_address, 'balanceOf', [USER]),
+            (proxy.contract_address, 'totalSupply', [])
```

Comment:
i guess this also speeds up testing?

Reply:
I forgot to compare! I'll confirm

Reply:
well it wasn't expected since the focus wasn't performance